compiler/rustc_passes/src/dead.rs

```diff
@@ -724,10 +790,58 @@ impl<'tcx> DeadVisitor<'tcx> {
                     );
                     err.note(&msg);
                 }
-                err.emit();
-            });
+                    err.emit();
+                },
+            );
+        }
+    }
+
+    fn warn_dead_fields_and_variants(&self, hir_id: hir::HirId, participle: &str, is_field: bool) {
+        if let Some(parent_node) = self.tcx.hir().find(hir_id)
```

Comment:
Instead of looking at the parent, why don't we pass it in? It seems like that you already have the item's info in the visitor.